src/core1_task.rs

```diff
@@ -111,7 +111,7 @@ pub fn wake_raspberry_pi(shared_i2c: &mut SharedI2C, delay: &mut Delay) -> bool
                 if let Ok(pi_is_awake) = shared_i2c.pi_is_awake_and_tc2_agent_is_ready(delay, false)
                 {
                     if pi_is_awake {
-                        break;
+                        return true;
```

Comment:
This doesn't seem correct to me – if the pi was already awake, then calling this function didn't wake it, and we'd want to be returning `false` from here, as we were previously, right?

Reply:
ah yes i had mis understood what the return value represents